src/dkg/round1.rs

```diff
@@ -239,6 +240,20 @@ impl PublicPackage {
     }
 }
 
+impl Ord for PublicPackage {
+    #[inline]
+    fn cmp(&self, other: &Self) -> cmp::Ordering {
+        Ord::cmp(&self.identity(), &other.identity())
+    }
+}
+
+impl PartialOrd<Self> for PublicPackage {
+    #[inline]
+    fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
```

Comment:
This is comparing the identity only, which can lead to situations where one has 2 different packages generated from the same identity, but with different contents, which compare as equal.

Would it be better to instead use `sort_unstable_by` in the `round2` code?

Reply:
Of course I meant [`sort_unstable_by_key`](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable_by_key), not `sort_unstable_by`

Reply:
That's a good point

Updated to use `sort_unstable_by_key`, since for now that's the only place that we need any ordering on PublicPackage